### ISSUE TITLE
[Bug] unpublished files can be viewed in the UI

### ISSUE THREAD

**Issue by HeshamElbashandy** [NCEAS/metacatui#976]
**Reproduce steps**: 

1. Create a dataset with few files and publish it (including publishing from the backend).
2. Edit the data package and replace one file with another file.
3. Log out 
4. View the published dataset and look at the Data Table, Image, and Other Data Details section.

**Expected result:**
The newly added data file shouldn't appear in any section in the data package until it is published from the backend. 

**Actual result:**
The newly added data file and the old data file appear in the Data Table, Image, and Other Data Details section.

**Comment by laurenwalker** [comment]
What do you mean by `publish from the backend`? Do you mean that you have a default access policy of `deny public read` but new data files added to an existing data package are set to `allow public read`?

**Comment by FiannaOBrien** [comment]
Non-public data files added to a published dataset should not be listed in the Data Table, Images, and Other Data section until they have been marked as public

**Comment by laurenwalker** [comment]
Gotcha, thanks.
